src/test/codewhisperer/service/keyStrokeHandler.test.ts

```diff
@@ -213,6 +214,62 @@ describe('keyStrokeHandler', function () {
             await KeyStrokeHandler.instance.processKeyStroke(mockEvent, mockEditor, mockClient, config)
             assert.ok(invokeSpy.called)
         })
+
+        it('Should skip invoking if there is immediate right context on the same line and not a single } for the user group', async function () {
+            const casesForSuppressTokenFilling = [
+                {
+                    rightContext: 'add',
+                    shouldInvoke: false,
+                },
+                {
+                    rightContext: '}',
+                    shouldInvoke: true,
+                },
+                {
+                    rightContext: '}    ',
+                    shouldInvoke: true,
+                },
+                {
+                    rightContext: ' add',
+                    shouldInvoke: true,
+                },
+                {
+                    rightContext: '    ',
+                    shouldInvoke: true,
+                },
+                {
+                    rightContext: '\naddTwo',
+                    shouldInvoke: true,
+                },
+            ]
+            casesForSuppressTokenFilling.forEach(async ({ rightContext, shouldInvoke }) => {
+                await testIfRightContextShouldInvoke(
+                    rightContext,
+                    shouldInvoke,
+                    CodeWhispererConstants.UserGroup.RightContext
+                )
+            })
+        })
+
+        it('Should not skip invoking based on right context for control group', async function () {
+            await testIfRightContextShouldInvoke('add', true, CodeWhispererConstants.UserGroup.Control)
```

Comment:
Thanks for de-duping this. Are there other cases (in old existing code) that could use `testIfRightContextShouldInvoke` ?

Reply:
yes I'll follow up with another PR to rector the other tests